app/screens/DashboardEditScreen.tsx

```diff
@@ -47,10 +57,21 @@ const renderSeparator = () => (
 export const DashboardEditScreen: FC<AppStackScreenProps<"EditDashboard">> =
   function DashboardEditScreen(_props) {
     const { navigation } = _props
-
+    const mutation = usePostDashboardPreferences()
     const [cards, setCards] = useState(initialCards)
-    const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>({})
 
+    const { dashboardPreferences } = useDashboardPreferences()
+    const initialPreferences = Object.entries(dashboardPreferences ?? {})
+      .filter(([_, value]) => value === true)
+      .reduce(
+        (acc, [key, value]) => {
+          acc[key] = value
+          return acc
+        },
+        {} as Record<string, boolean>,
+      )
+
+    const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>(initialPreferences)
     const toggleChecked = (id: string, newChecked: boolean) => {
       setCheckedStates((prev) => ({ ...prev, [id]: newChecked }))
     }
```

Comment:
I get the idea + you're also just using code that was already available to you so I understand why you did this, but currently to do this POST we go from type `DashboardPreferences` -> `Record<string, boolean>` -> `DashboardPreferences`

I think the better thing to do here is to just stick with the `DashboardPreferences` shape the whole time so we dont need to double transform.

Edit: I also think that we should consider the default state as all checked right? and then the user can deselect as they so desire

```suggestion
    const [checkedStates, setCheckedStates] = useState<<DashboardPreferences>(
      dashboardPreferences ?? {
        whos_on_duty: true,
        upcoming_shifts: true,
        upcoming_leaves: true,
        open_shifts: true,
        team_roster: true,
      },
    )
    const toggleChecked = (id: keyof DashboardPreferences, checked: boolean) => {
      setCheckedStates((prev) => ({ ...prev, [id]: checked }))
    }
```